designs/comprehensions-design.md

```diff
@@ -0,0 +1,105 @@
+# Comprehensions Team Design Doc
+
+## Test cases
+
+```
+a : List = None
+a = [i for i in range(10)]
+print(a)
+```
+This should output ```[0,1,2,3,4,5,6,7,8,9]```
+
+```
+a : List = None
+j : int = 9
+a = [j for i in range(5)]
```

Comment:
Note that some of what's required here is similar to closures (in order to get references to `j` to work out)